src/euphorie/client/navigation.py

```diff
@@ -29,11 +29,29 @@ def FindFirstQuestion(dbsession, filter=None):
 
 
 def FindNextQuestion(after, dbsession, filter=None):
+    # Filter out Choice objects where condition is not met,
+    # i.e. either tree item is not a choice,
+    # or it is a Choice and the condition is blank,
+    # or an option that is referred to in the condition has been picked
+    # XXX simplify?
+    condition = sql.or_(
+        ~sql.exists().where(model.Choice.id == model.SurveyTreeItem.id),
+        sql.exists().where(
+            sql.and_(
+                model.Choice.id == model.SurveyTreeItem.id,
+                sql.or_(
+                    model.Choice.condition == None,  # noqa: E711
+                    model.Choice.condition.like(model.Option.zodb_path),
```

Comment:
This is unclear to me...
You are using `like`, but AFAIK `like` requires a pattern to match.
Is `model.Choice.condition.like(model.Option.zodb_path)` containing an `%`?
If not, it seems to me that this is equivalent to:
```
model.Choice.condition == model.Option.zodb_path
```
If yes, it would be unexpected for me.